examples/helm/tiller-template.yaml

```diff
@@ -58,6 +73,8 @@ objects:
         containers:
         - name: tiller
           image: gcr.io/kubernetes-helm/tiller:${HELM_VERSION}
+          securityContext:
+            runAsUser: 1001
```

Comment:
no normal user has this permission by default, is it really required?

Reply:
Yes because the image of tiller set USER at "nobody" and the deployment fails. 
https://github.com/helm/helm/blob/master/rootfs/Dockerfile#L25

If you have a better idea, I'm interested.

Reply:
it doesn't matter what the dockerfile sets the USER to.  when openshift runs the image, it will assign an arbitrary uid.

were you seeing a failure that lead you to add this?

Reply:
I have this error : 

```Error: container has runAsNonRoot and image has non-numeric user (nobody), cannot verify user is non-root```

Reply:
when do you see that error?  doing what?

Reply:
Hello,

Here the complete action and results : 



```
$ export TILLERNAME_SPACE=helm-test-project
$ oc new-project helm-test-project
$ oc process -p TILLER_NAMESPACE=$TILLERNAME_SPACE \
    -p HELM_VERSION=v2.11.0 -f tiller-template.yaml > tiller.yaml
$ oc apply -f tiller.yaml                           
serviceaccount "tiller" created
rolebinding "tiller" created
deployment "tiller" created
$ oc get pods
NAME                      READY     STATUS                       RESTARTS   AGE
tiller-555c89d54d-5l5nb   0/1       CreateContainerConfigError   0          14s
$ oc logs tiller-555c89d54d-5l5nb      
Error from server (BadRequest): container "tiller" in pod "tiller-555c89d54d-5l5nb" is waiting to start: CreateContainerConfigError
$  oc get events       
LAST SEEN   FIRST SEEN   COUNT     NAME                                       KIND         SUBOBJECT                 TYPE      REASON                  SOURCE                                MESSAGE
3m          3m           1         tiller-555c89d54d-5l5nb.155ce4312edbcdf0   Pod                                    Normal    Scheduled               default-scheduler                     Successfully assigned tiller-555c89d54d-5l5nb to phmbucoacp572.example.com
3m          3m           1         tiller-555c89d54d-5l5nb.155ce431404234d1   Pod                                    Normal    SuccessfulMountVolume   kubelet, phmbucoacp572.example.com   MountVolume.SetUp succeeded for volume "tiller-token-txshr" 
2m          3m           11        tiller-555c89d54d-5l5nb.155ce431ab2c9aa8   Pod          spec.containers{tiller}   Normal    Pulled                  kubelet, phmbucoacp572.example.com   Container image "gcr.io/kubernetes-helm/tiller:v2.11.0" already present on machine
2m          3m           11        tiller-555c89d54d-5l5nb.155ce431ab493a27   Pod          spec.containers{tiller}   Warning   Failed                  kubelet, phmbucoacp572.example.com   Error: container has runAsNonRoot and image has non-numeric user (nobody), cannot verify user is non-root
3m          3m           2         tiller-555c89d54d-5l5nb.155ce431cc25bf68   Pod                                    Normal    SandboxChanged          kubelet, phmbucoacp572.example.com   Pod sandbox changed, it will be killed and re-created.
3m          3m           1         tiller-555c89d54d.155ce4312dedb505         ReplicaSet                             Normal    SuccessfulCreate        replicaset-controller                 Created pod: tiller-555c89d54d-5l5nb
3m          3m           1         tiller.155ce4312c2d7bc6                    Deployment                             Normal    ScalingReplicaSet       deployment-controller                 Scaled up replica set tiller-555c89d54d to 1
```

Reply:
This worked for me with the current template (none of your changes)
```
$ oc new-project helm
$ oc process -p TILLER_NAMESPACE=helm -p HELM_VERSION=v2.11.0 -f examples/helm/tiller-template.yaml | oc apply -f -
$ oc get pods
NAME                      READY     STATUS    RESTARTS   AGE
tiller-6b65f4fbd7-rgwxf   1/1       Running   0          1m
```

Reply:
and the pod yaml shows:
```
   securityContext:
      capabilities:
        drop:
        - KILL
        - MKNOD
        - SETGID
        - SETUID
      runAsUser: 1000100000
```

so i don't know why yours seems to be set to "runAsNonRoot"

Reply:
I think our cluster have differents scc.

Reply:
my cluster uses the standard defaults, I don't want to customize this template for your cluster configuration.